src/Cutscenes.ts

```diff
@@ -1637,6 +1637,193 @@ export class Cutscenes<TEightBittr extends FullScreenPokemon> extends EightBittr
+        // Make defending pokemon flicker
+        this.EightBitter.TimeHandler.addEvent(
+            (): void => {
+                attacker.hidden = !attacker.hidden;
+                this.EightBitter.animations.animateFlicker(defender, 12, 6, args.callback);
+            },
+            40);
+
+        /*const animateExplosion: (x: number, y: number) => void = (x: number, y: number): void => {
+            const explosion: IThing = this.EightBitter.ObjectMaker.make("ExplosionLarge");
+            this.EightBitter.things.add(explosion, x, y);
+            /*this.EightBitter.TimeHandler.addEvent(
+                (): void => {
+                    this.EightBitter.physics.killNormal(explosion);
+                },
+                20 + 10);
+            console.log(x);
+            console.log(y);
+        };*/
+
+        const explosions: IThing[] = [
+            this.EightBitter.ObjectMaker.make("ExplosionLarge"),
+            this.EightBitter.ObjectMaker.make("ExplosionLarge"),
+            this.EightBitter.ObjectMaker.make("ExplosionLarge")
+        ];
+        let startX: number[] = [];
+        let startY: number[] = [];
+        if (direction === -1) {
+            startX[0] = menu.right - defender.width / 2 * this.EightBitter.unitsize;
+            startY[0] = menu.top;
+        } else {
+            console.log(direction);
+            startX[0] = menu.left + (defender.width + 32) * this.EightBitter.unitsize;
+            startY[0] = menu.bottom - (defender.height + 16) * this.EightBitter.unitsize;
+            startX[1] = startX[0] + 6 * this.EightBitter.unitsize;
+            startY[1] = startY[0] - 6 * this.EightBitter.unitsize;
+            startX[2] = startX[1] + 6 * this.EightBitter.unitsize;
+            startY[2] = startY[1] - 8 * this.EightBitter.unitsize;
+        }
+
+        const animateExplosions: () => void = (): void => {
+            this.EightBitter.TimeHandler.addEvent(
+                (): void => {
+                    this.EightBitter.things.add(explosions[0], startX[0], startY[0]);
+                },
+                0);
+            this.EightBitter.TimeHandler.addEvent(
+                (): void => {
+                    this.EightBitter.physics.killNormal(explosions[0]);
+                    this.EightBitter.things.add(explosions[1], startX[1], startY[1]);
+                },
+                4);
+            this.EightBitter.TimeHandler.addEvent(
+                (): void => {
+                    this.EightBitter.physics.killNormal(explosions[1]);
+                    this.EightBitter.things.add(explosions[2], startX[2], startY[2]);
+                },
+                8);
+            this.EightBitter.TimeHandler.addEvent(
+                (): void => {
+                    this.EightBitter.physics.killNormal(explosions[2]);
+                },
+                12);
+        };
+
+        this.EightBitter.TimeHandler.addEvent(
+                (): void => {
+                    animateExplosions();
```

Comment:
Why make a separate function named `animateExplosions` if you only call it once? Put all the logic in this one. Less complication.